pkg/auth/providers/googleoauth/goauth_provider.go

```diff
@@ -401,12 +401,14 @@ func (g *googleOauthProvider) getDirectoryService(ctx context.Context, userEmail
 		// using JWTConfigFromJSON method
 		config, err := google.JWTConfigFromJSON(jsonCredentials, admin.AdminDirectoryUserReadonlyScope, admin.AdminDirectoryGroupReadonlyScope)
 		if err != nil {
-			return nil, err
+			logrus.Error(err)
+			return nil, fmt.Errorf("invalid Service Account Credentials provided")
```

Comment:
seems like the func returns error for other reasons too? can we parse error to make sure?

Reply:
`JWTConfigFromJSON` returns error either if parsing service account credentials fails, or if it's not of the type service account credential. So the reason is common for both that the service account creds are invalid.